### PR TITLE
feat: Rustify lempel_ziv_complexity

### PR DESCRIPTION
**Pull request by abstractqqq**
Implemented the first Polars Plugin: lempel_ziv_complexity. We now have both eager and lazy version for this feature too ( and is super fast )

How to test:
1. maturin develop --release
2. If package is not install in local environment, do pip install .
3. pytest tests/test_tsfresh (there is a test on lempel_ziv_complexity, which is the expr plugin implemented in this PR)

Definitely need to review this process and try to improve this process. @topher-lo

### CODE REVIEW

**Comment by ritchie46** [comment]
@messense can you shed some light on how we can ensure this best?

I thought maturin build always put the shared library in the same folder as the python code?

**Comment by abstractqqq** [comment]
> @abstractqqq I expect you have your file names incorrect.
> 
> It is important that the name in the Cargo.toml, the python package and the rust lib match.

I think the file isn't generated in the right location on windows... I did the same thing and got it working on Linux. It is working now

**Comment by messense** [comment]
> I thought maturin build always put the shared library in the same folder as the python code?

By default yet, but it can be changed via `[tool.maturin] module-name` in `pyproject.toml`

https://github.com/neocortexdb/functime/blob/14fa0813c524f23032947a30862551fd2fb6c957/pyproject.toml#L131

**Comment by topher-lo** [comment]
@abstractqqq Apart from the conflicts (nothing too serious), I think this is good to go.

**Review by topher-lo — APPROVED** [review]

File: src/feature_extraction/feature_extractor.rs [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
+        |op_b| op_b.unwrap_or(false)
+    ).collect();
 
     let mut ind:usize = 0;
     let mut inc:usize = 1;
 
-    let mut sub_strings: HashSet<&[u8]> = HashSet::new();
-    while ind + inc <= s.len() {
-        let subseq: &[u8] = &s[ind..ind+inc];
+    let mut sub_strings: HashSet<&[bool]> = HashSet::new();
```
Comment (ritchie46) [inline src/feature_extraction/feature_extractor.rs:23]: @abstractqqq rusts default hastable/hashset has a very expensive hash function.

I recommended using hashbrown here.

Comment (topher-lo) [reply src/feature_extraction/feature_extractor.rs:23]: Resolved in #113

**Pull request merged**